### ISSUE TITLE
Errors building MonoGame 3.6 on Windows 10 using Visual Studio 2017

### ISSUE THREAD

**Issue by davidsummers** [MonoGame/MonoGame#6169]
I'm trying to build MonoGame itself on Windows 10 with Visual Studio 2017 so I can try to figure out why my application is showing a NullReferenceException in the Game class.

The documentation says to run Protobuild.exe and then you should be able to build MonoGame from source.

I do that and it completes without error.

Then I load Visual Studio using the MonoGame.Framework.Windows.sln solution.

When I try to build MonoGame.Framework.Windows project, I get these errors:

1>------ Build started: Project: MonoGame.Framework.Windows, Configuration: Debug Any CPU ------
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Graphics\Vertices\VertexBuffer.cs(80,1,80,1): warning CS1570: XML comment has badly formed XML -- 'Expected an end tag for element 'param'.'
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\MediaPlayer.WMS.cs(15,24,15,36): error CS0246: The type or namespace name 'MediaSession' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\MediaPlayer.WMS.cs(16,24,16,41): error CS0246: The type or namespace name 'AudioStreamVolume' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\Song.WMS.cs(16,18,16,26): error CS0246: The type or namespace name 'Topology' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\Video.WMS.cs(10,18,10,26): error CS0246: The type or namespace name 'Topology' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\Video.WMS.cs(9,17,9,25): error CS0246: The type or namespace name 'Topology' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\Song.WMS.cs(14,17,14,25): error CS0246: The type or namespace name 'Topology' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\VideoPlayer.WMS.cs(14,24,14,36): error CS0246: The type or namespace name 'MediaSession' could not be found (are you missing a using directive or an assembly reference?)
1>C:\Projects\MonoGame.orig\MonoGame.Framework\Media\VideoPlayer.WMS.cs(15,24,15,41): error CS0246: The type or namespace name 'AudioStreamVolume' could not be found (are you missing a using directive or an assembly reference?)
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

It appears that some other setup is needed to build MonoGame other than just running Protobuild?

Is this documented somewhere?  I've searched documentation for a couple of days and so far I just see documentation that says run Protobuild.

**Comment by davidsummers** [comment]
I just tried the latest 'develop' branch instead of the 'v3.6' tag and the 'develop' branch compiles fine.

Does that mean there is a bug in the build of v3.6 tag on Windows 10 with Visual 2017?

Is this fixed in v3.6.1? If so, when will v3.6.1 be released?

**Comment by DDReaper** [comment]
This is a known issue in 3.6.  So you have to use the latest dev 3.7 release.  Either from the download page, or source

**Comment by davidsummers** [comment]
OK, thanks for the information.

Is there a pre-built DEBUG version of MonoGame 3.6 for Windows (DirectX) that I could use from someone?

I've got two programs that each create my class which subclasses Game and one is working and one is crashing on a NullReferenceException error somewhere in the Game constructor.

At the moment I don't see any differences between the two programs but If I could step into it I might be able to figure out the difference and fix the crashing program to not crash.

**Comment by DDReaper** [comment]
Best way to debug, is to clone the source locally, add the source to your project and then remove the MonoGame references in your project and replace them with the MG projects.

Once you have cloned the project, just Run "ProtoBuild.exe" to generate the sln / csproj project files.

**Comment by davidsummers** [comment]
When you talked about "add the source to your project" did you mean the MonoGame v3.6 MonoGame.Framework.Windows project in the GitHub MonoGame project?

I cloned the MonoGame project from GITHUB and then ran the Protobuild.exe.

The new part is that I've now added the MonoGame.Framework.Windows to my mgtest project and those are the only two projects in my "solution".

When I try to build the DEBUG MonoGame.Framework.Windows project I get the same errors as above.

If you were talking about the 3.7 DEV then there was a mis-understanding.  I am trying to figure out if it is possible and how I could debug into a 3.6 DEBUG version of MonoGame.

I'm a newbie on C# so hopefully I am either just mis-understanding what you said or did something wrong.

My test program can be found at https://github.com/davidsummers/mgtest URL.

Please advise.  Thanks.

**Comment by DDReaper** [comment]
OK, breaking the instructions down a bit (hope that helps)

1. Clone Monogame to your PC
2. Run ProtoBuild to generate the sln / csproj files in MonoGame
3: Open your game project
4: In the Solution Explorer, right-click the Solution and select "Add existing project"
5. Browse to the csproj file for the Monogame.Framework.WindowsUniversal project (click OK to add)
6. Expand your project in Solution explorer and then expand the "References"
7. Right-click and remove the existing MonoGame.Framework reference
8. Right click the References in your project and select add reference
9. Ensure you have selected the "Solution" branch under Projects and select MonoGame.Framework.Universal.

You now have a project using the Monogame source and not the pre-packaged references.
And you should now be able to catch when the project breaks when you run it in Debug mode.

Hope that helps.

**Comment by davidsummers** [comment]
@DDReaper : Well, I got a bit further.   I was able to add the MonoGame.Framework.WindowsUniversal project to my solution.   Then I deleted the existing reference to Monogame in the project references.  Then I went to add the new MonoGame reference on your step 7.  However, when I click the OK button, an error dialog comes that says "Microsoft Visual Studio -> Unable to add a reference to project 'MonoGame.Framework.WindowsUniversal'.

However, I *CAN* now compile the MonoGame.Framework.WindowsUniversal project in my solution whereas before when I was just using the MonoGame.Framework.Windows I got the errors listed in my first comment above.

So, one step further.  Now the question is: Why won't it allow me to add that reference as detailed in your step 7?

**Comment by tomspilman** [comment]
@DDReaper @davidsummers - Is this issue complete?  Can i close it or do we have more to work out here?

**Comment by davidsummers** [comment]
I finally found the problem.  The problem was that somehow we had gotten different configuration references to MonoGame in our different projects in our Visual Studio solution.  One project was crashing on MonoGame initialization and others weren't, which was really confusing because we were initializing them the same way.

Once we set all the MonoGame references to be the same (correct) one then all our crash problems went away.

Sorry for the false alarm, the problem was on our side.

Thanks for the help.

This issue can be closed.
